### PR TITLE
Support for hash functions with length other than 20 bytes

### LINKED ISSUE #168: Optionally supporting custom hash function (and id space size)
I have an idea to use the same implementation as Bittorrent DHT, but with bigger id space, namely using 256bits sha2-256 or sha3-256 instead of 160bits sha1.
Support for this can be implemented as an optional argument with custom hash function.

Would you like to accept a PR that implements this feature?

I understand that this is not in the scope of this project, but forking the whole things just for such small feature doesn't seem to be reasonable.

### PR DESCRIPTION
**Pull request by nazar-pc**
Do not use hardcoded hash length, but rather determine it based on supplied hash function

Fixes #168
Depends on https://github.com/tristanls/k-bucket/pull/42, https://github.com/mafintosh/k-rpc/pull/11

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -26,7 +26,9 @@ function DHT (opts) {
   this._peers = new PeerStore(opts.maxPeers || 10000)
 
   this._secrets = null
-  this._rpc = krpc(opts)
+  this._hash = opts.hash || sha1
+  this._hash_length = this._hash(Buffer.from('')).length
```
Comment (feross) [inline client.js]: We use `camelCase` in this project. The leading underscore only indicates a private variable. Can you rename this to `this._hashLength`?

File: client.js [hunk 2]
```diff
@@ -26,7 +26,9 @@ function DHT (opts) {
...
+  this._rpc = krpc(Object.assign({idSpace: this._hash_length}, opts))
```
Comment (feross) [inline client.js]: `idSpace` -> `idLength`

**Comment by nazar-pc** [comment]
Changed as requested and rebased to resolve merge conflicts

**Comment by feross** [comment]
Released as 7.7.0

**Pull request merged**